main.tf

```diff
@@ -5,6 +5,7 @@ locals {
   kms_key_pattern = format("arn:aws:kms:%s:%s:key/*", local.xo_account_region, var.xo_account_id)
   s3_bucket = "xosphere-io-releases-${data.aws_region.current.name}"
   xo_account_region = "us-west-2"
+  has_global_terraform_settings = var.terraform_version != "" || var.terraform_aws_provider_version != "" || var.terraform_backend_aws_region != "" || var.terraform_backend_s3_bucket != "" || var.terraform_backend_s3_key != ""
```

Comment:
should this be an AND instead of an OR?

Reply:
The reason to have OR instead of AND is that "has_global_terraform_settings" is used to determine if we have either of those set from the global settings. If there is no global settings, and there are no tags from the instance related to terraform, we will skip the terraform operation.

Reply:
Ok, so we install the lambdas all of the time, and then only call them under certain conditions?  My concern here is that we may end up calling the lambda when these options aren't set.  I guess we would need some assurance that either:
a) The conditions for Invoke()'ing this Lambda is an AND
or
b) This Lambda verifies inputs and logs/early exit when required params are not provided.

Reply:
Yes, we have both of what you mentioned.

Reply:
Cool.  LGTM